src/App.tsx

```diff
@@ -1,24 +1,179 @@
+  useCallback,
+} from 'react';
+import { TodoService } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { filterTodos } from './utils/helpers';
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+import { AddingForm } from './components/AddingForm';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { NotificationError } from './components/NotificationError';
+import { Errors } from './types/Errors';
+
+const USER_ID = 6922;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoStatus, setTodoStatus] = useState<TodoStatus>(TodoStatus.ALL);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState<boolean>(false);
+  const [error, setError] = useState<Errors>(Errors.NONE);
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const loadedTodos = await TodoService.getByUserId(USER_ID);
+
+      setTodos(loadedTodos);
+    } catch (err) {
+      setError(Errors.LOAD);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todos, todoStatus);
+  }, [todos, todoStatus]);
+
+  const numberOfRemainingTodos = useMemo(() => {
+    return filterTodos(todos, TodoStatus.ACTIVE).length;
+  }, [todos]);
+
+  const addTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    setIsInputDisabled(true);
+    const temporaryTodo = {
+      id: 0,
+      ...todoData,
+    };
+
+    try {
+      setTempTodo(temporaryTodo);
+      setLoadingTodosIds(currIds => [...currIds, temporaryTodo.id]);
+      const addedTodo = await TodoService.create(temporaryTodo);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (err) {
+      setError(Errors.ADD);
+    } finally {
+      setTempTodo(null);
+    }
+
+    setIsInputDisabled(false);
+  }, []);
+
+  const removeTodo = useCallback(async (todoIds: number[]) => {
+    setLoadingTodosIds(currIds => [...currIds, ...todoIds]);
+
+    try {
+      await Promise.all(todoIds.map((todoId) => TodoService.delete(todoId)));
+
+      setTodos(currTodos => currTodos.filter(
+        (todo) => !todoIds.includes(todo.id),
+      ));
+    } catch (err) {
+      setError(Errors.DELETE);
+    } finally {
+      setLoadingTodosIds([]);
+    }
+  }, []);
+
+  const updateTodoData = useCallback(async (
+    todoId: number,
+    updatedTodo: Partial<Todo>,
+  ) => {
+    setLoadingTodosIds((currIds) => [...currIds, todoId]);
+
+    try {
+      await TodoService.update(todoId, updatedTodo);
+      setTodos(currTodos => currTodos.map(todo => (todo.id === todoId
+        ? {
+          ...todo,
+          ...updatedTodo,
+        }
+        : todo)));
```

Comment:
better rename currTodos to todos, you can also destructure properties of todo & updatedTodo on the same line, this will look much shorter and better, as for me
better put todo.id === todoId condition on the new line, for better readability

Reply:
It`s just because I use "todos" as a name of variable in the state so I decided to change it that way